File: logineer/src/main/java/br/com/gabryel/logineer/service/UserService.java
package br.com.gabryel.logineer.service;

import br.com.gabryel.logineer.dto.UserDto;
import br.com.gabryel.logineer.entities.User;

public interface UserService {

    User register(UserDto user);

    User login(String email, String password);

    User getUser(String id, String token);

    User convertToUser(UserDto userDto);
}

File: logineer/src/main/kotlin/br/com/gabryel/logineer/service/UserService.kt
package br.com.gabryel.logineer.service

import br.com.gabryel.logineer.dto.UserDto
import br.com.gabryel.logineer.entities.User
import br.com.gabryel.logineer.exceptions.LogineerException

interface UserService {

    @Throws(LogineerException::class)
    fun register(userDto: UserDto): User

    @Throws(LogineerException::class)
    fun login(email: String, password: String): User

    @Throws(LogineerException::class)
    fun getUser(id: String, token: String?): User

    fun convertToUser(userDto: UserDto): User
}